app/src/main/java/com/example/learn09082021v001/ReadAndWriteMemory.kt

```diff
@@ -9,25 +9,26 @@ import java.io.BufferedReader
 import java.io.FileInputStream
 import java.io.InputStreamReader
 
-class ReadAndWriteMemori {
-    var dataMemoriLine1 = ""
-    var dataMemoriLine2 = ""
-    var dataMemoriLine3 = ""
-    var dataMemoriLine4 = ""
-    var dataMemoriLine5 = ""
-    var dataMemoriLine6 = ""
-    var dataMemoriLine7 = ""
+class ReadAndWriteMemory {
+    var dataMemoryLine1 = ""
+    var dataMemoryLine2 = ""
+    var dataMemoryLine3 = ""
+    var dataMemoryLine4 = ""
+    var dataMemoryLine5 = ""
+    var dataMemoryLine6 = ""
+    var dataMemoryLine7 = ""
 
-    fun WriteDataMemory(
+    fun writeDataMemory(
         context: Context,
         fileNameSaverMemory: String,
-        dataSaverMemori: String = "",
+        dataSaverMemory: String = "",
     ) {
         val fileOutputStream: FileOutputStream
         try {
             fileOutputStream = context.openFileOutput(fileNameSaverMemory, Context.MODE_PRIVATE)
-            fileOutputStream.write(dataSaverMemori.toByteArray())
+            fileOutputStream.write(dataSaverMemory.toByteArray())
             Toast.makeText(context, "data update complete", Toast.LENGTH_SHORT).show()
+            fileOutputStream.close()
```

Comment:
the `close` have to get call when exception as well, so only close here is not enough
you can add this function on `finally